pkg/apis/kubernetes/v1/cluster_types.go

```diff
@@ -0,0 +1,33 @@
+package v1
+
+import (
+	"context"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/apis/internal"
+)
+
+// anxcloud:object
+
+// Cluster represents a Kubernetes cluster
+// This resource does not support updates
+type Cluster struct {
+	internal.CommonMethods
+	State *internal.State `json:"state,omitempty"`
```

Comment:
do generic services specify the state always being stored in the attribute `State`? We could then add it to `internal.CommonMethods`/`gs.GenericService` instead - at least when we don't need a API-specific `State` type for adding the state values ..

Reply:
Generic Service resource definitions have to include one attribute of type State, but it does not have to be called `state`.